src/tools/rustfmt/src/items.rs

```diff
@@ -2395,6 +2395,33 @@ fn rewrite_explicit_self(
                 )?),
             }
         }
+        ast::SelfKind::Pinned(lt, m) => {
+            let mut_str = m.ptr_str();
+            match lt {
+                Some(ref l) => {
+                    let lifetime_str = l.rewrite_result(
+                        context,
+                        Shape::legacy(context.config.max_width(), Indent::empty()),
+                    )?;
+                    Ok(combine_strs_with_missing_comments(
+                        context,
+                        param_attrs,
+                        &format!("&{lifetime_str} pin {mut_str} self"),
+                        span,
+                        shape,
+                        !has_multiple_attr_lines,
+                    )?)
+                }
+                None => Ok(combine_strs_with_missing_comments(
+                    context,
+                    param_attrs,
+                    &format!("&pin {mut_str} self"),
+                    span,
+                    shape,
+                    !has_multiple_attr_lines,
+                )?),
+            }
```

Comment:
only compute the `lifetime_str` (or the entire format since that is simpler) in the match and run the logic that is the same outside of it

Reply:
This code is essentially a copy of the code in the earlier match arm for `Region`.  If we're reworking this, probably it'd also make sense to pull out this shared logic in a way that both can use.

Reply:
Ah yea. I would prefer that. I know it's preexisting but I'd like to see that made reusable in a separate commit. It's fine to do it after the current commit, no need to do it before or in a separate PR